yb_migrate/src/utils/diagnostics.go

```diff
@@ -0,0 +1,143 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/fs"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+//call-home json formats
+var (
+	jsonFilePath    string
+	sendReq         sendRequest
+	jsonBuf         []byte
+	SendDiagnostics bool
+)
+
+type sendRequest struct {
```

Comment:
If you name the struct as payload then you can make some of these function as method of that struct like: `Send` instead of `SendPayload()` , `Get` instead of `GetPayload`. There you can this struct as public.

I am suggestion something like this:

```
type struct Payload {
....
}

func (payload *Payload) Send() {}

func (payload *Payload) Get() {} 

func (payload *Payload) Init() {} 

```

Reply:
Keeping the struct public would destroy the purpose of using an explicit getter method, but I have renamed for readability purposes

Reply:
yeah ignore `type struct Payload` i meant `type struct payload`